### DISCUSSION TITLE
berry scripting - get var into sensor value

### DISCUSSION THREAD

**Discussion by StefanTrethan** [arendst/Tasmota#18194] [Q&A] [answered]
I'm completely new to berry so please excuse my basic question, but how do a get a value from a berry script to exist in Tasmota as a sensor value? I mean so it is shown on the web interface, and so I can use it as a condition for rules.

Background is I need to fetch exported power from another wifi device, which I already managed with this berry script:

cl = webclient()
cl.begin("http://10.0.0.133/public")
r = cl.GET()
s = cl.get_string()
import json
dict = json.load(s)
var pow = 60 
pow = dict['data']['tblDTSU'][0]['em_Pt']
print(pow)

This works and the current numeric value of pow is correctly printed out in the console. I still need to make it happen periodically, but I believe I can manage that (have not tried yet). 
However no matter what I try I can't seem to get this value out of the berry scripting console.
I would expect "web_sensor(): display sensor information on the Web UI" may be the key, but I can't get that to work or find an example on how to use it.

Thanks for your help.

Stefan

**Comment by sfromis** [comment, marked as answer]
What you can do is to create a driver in Berry, which gets the opportunity of supplying sensor data to both Tasmota front page and JSON for the regular SENSORS payloads. 
Here's an example: https://tasmota.github.io/docs/Berry-Cookbook/#full-example
For your purpose, you can ignore the I2C-related parts, the key for you is the functions `web_sensor()` and `json_append()`, and that it is done in a class, of which an instance is then added via `tasmota.add_driver()`

Also see: https://tasmota.github.io/docs/Berry/#creating-a-tasmota-driver

**Comment by StefanTrethan** [comment]
Thank you, that worked fine.
